### PR TITLE
tests: update execution-spec-tests to 1.0.5

### PR DESCRIPTION
**Pull request by holiman**
This updates the execution-spec-tests to latest version. Putting it up to see if it builds

### CODE REVIEW

**Review by lightclient — APPROVED** [review]

**Comment by holiman** [comment]
So the tests passes, yay ... however, we don't actually execute the cancun stuff
```
	bt.skipLoad(`^cancun/`)
	bt.skipLoad(`-fork=Cancun`)
```
So let's not merge this until we get that working

**Comment by holiman** [comment]
```
[user@work go-ethereum]$ go run   ./cmd/evm/ --json  blocktest ./tests/spec-tests/fixtures/cancun/eip5656_mcopy/mcopy_memory_expansion/mcopy_huge_memory_expansion.json

```
```
...
{"pc":47,"op":94,"gas":"0x5dd9e6e","gasCost":"0x3","memSize":256,"stack":["0x0","0x0","0x1","0x0","0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"],"depth":2,"refund":0,"opName":"MCOPY","error":"gas uint64 overflow"}
{"pc":38,"op":85,"gas":"0x17d5d2","gasCost":"0x5654","memSize":256,"stack":["0x1","0x0"],"depth":1,"refund":0,"opName":"SSTORE"}
{"pc":39,"op":0,"gas":"0x177f7e","gasCost":"0x0","memSize":256,"stack":[],"depth":1,"refund":0,"opName":"STOP"}
...
########## BAD BLOCK #########
Block: 1 (0x0133b010e20067d6f0c1fac34c8245b8968d87d968ca3096ef8eeeb14d7e92c0)
Error: invalid merkle root (remote: 27557b93e5bb671d712b61c0da570f6883abbc7bd2da95f0af4e462f139f41bb local: 1051394ef6cb6fbcc2838024c0ca9ff6ca975ef4aefc041d82608ce36f44e7c7) dberr: %!w(<nil>)

```

**Comment by holiman** [comment]
This is the (complete) state that I get
```json
{
    "root": "1051394ef6cb6fbcc2838024c0ca9ff6ca975ef4aefc041d82608ce36f44e7c7",
    "accounts": {
        "0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b": {
            "balance": "10779762",
            "nonce": 1,
            "root": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
            "codeHash": "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
        }
    }
}
```

File: build/checksums.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # This file contains sha256 checksums of optional build dependencies.
 
-# version:spec-tests 1.0.2
+# version:spec-tests 1.0.5
 # https://github.com/ethereum/execution-spec-tests/releases
-# https://github.com/ethereum/execution-spec-tests/releases/download/v1.0.2/
-24bac679f3a2d8240d8e08e7f6a70b70c2dabf673317d924cf1d1887b9fe1f81  fixtures.tar.gz
+# https://github.com/ethereum/execution-spec-tests/releases/download/v1.0.5/
+322085c2c2e173069d38d79034518e91bd53b29eaafd92a56f0ef753b5336aa7  fixtures.tar.gz
```
Comment (danceratopz) [inline build/checksums.txt:6]:
```suggestion
d4fd06a0e5f94beb970f3c68374b38ef9de82d4be77517d326bcf739c3cbf3a2  fixtures_develop.tar.gz
```

**Review by danceratopz — COMMENTED** [review]
Hi @holiman, execution-spec-tests artifact naming changed in v1.0.5. The name of the fixtures file to download in order to get fixtures for features under development is now:
```
fixtures_develop.tar.gz
```

L349 needs to be changed accordingly here:
https://github.com/MariusVanDerWijden/go-ethereum/blob/master/build/ci.go#L347-L349

And the change below is also required.

Sorry about that! More info in "breaking changes" in the release notes:
https://github.com/ethereum/execution-spec-tests/releases/tag/v1.0.5

**Comment by holiman** [comment]
Thanks @danceratopz !

**Comment by holiman** [comment]
Now the failing ones are 

- `cancun/eip4788_beacon_root/beacon_root_contract/beacon_root_precompile_calls.json` (`000`-`011`), 
- `cancun/eip4788_beacon_root/beacon_root_contract/beacon_root_precompile_timestamps.json` (`000`-`023`) and
- `cancun/eip4788_beacon_root/blocks_beacon_root_contract/beacon_root_transition_test.json`

**Comment by holiman** [comment]
The last one fails to import block 15, with
```
Block: 15 (0x15eabba755d11877f1e4da90d93c0805d33472521991b795a497093a43ec1caf)
Error: invalid gas used (remote: 166464 local: 124548)
```
One thing I noticed was that it calls `0xb`:
```
{"pc":22,"op":241,"gas":"0xeef7f","gasCost":"0x190cb","memSize":32,"stack":["0x20","0x20","0x20","0x0","0x0","0xb","0x186a0"],"depth":1,"refund":0,"opName":"CALL"}
```
Is that maybe a remnant from when the beacon root was at `0xb`?

**Comment by danceratopz** [comment]
> Now the failing ones are
> 
>     * `cancun/eip4788_beacon_root/beacon_root_contract/beacon_root_precompile_calls.json` (`000`-`011`),
> 
>     * `cancun/eip4788_beacon_root/beacon_root_contract/beacon_root_precompile_timestamps.json` (`000`-`023`) and
> 
>     * `cancun/eip4788_beacon_root/blocks_beacon_root_contract/beacon_root_transition_test.json`

> Is that maybe a remnant from when the beacon root was at 0xb?

@holiman, looking at the names and the fail, it seems to me that these are indeed leftover artifacts from v1.0.2 that refer to an out-date beacon root spec. Could it be that the folder that the CI downloads fixtures into isn't cleaned between runs?

**Comment by holiman** [comment]
@danceratopz aargh, right, again: thanks!

**Pull request merged**